src/video-player.js

```diff
@@ -274,11 +278,32 @@ class VideoPlayer extends Utils.mixin(Eventable) {
     this.videojs.ready(() => {
       onReady();
 
+
       if (ready) {
         ready(this);
       }
     });
 
+
+    this.nbCalls = 0;
+    this.reTryVideo = (maxNumberOfCalls, timeout) => {
+      if (!this.isVideoReady()) {
```

Comment:
Based on the previous comment - the fact that we can modify the video player's source in any given time means that we need to make sure to reset the timers here on such event, otherwise weird behaviour may occur.